app/design-system/design-system-hedvig/src/main/kotlin/com/hedvig/android/design/system/hedvig/Tab.kt

```diff
@@ -0,0 +1,443 @@
+            TabItem(
+              modifier = Modifier
+                .clip(tabSize.tabShape),
+              onClick = {
+                onTabChosen(index)
+              },
+              text = title,
+              textStyle = tabSize.textStyle,
+              tabTextColor = tabStyle.colors.textColor,
+              contentPadding = tabSize.tabPadding,
+            )
+          }
+        },
+      ),
+    ) { allMeasurables: List<List<Measurable>>, constraints ->
+      val tabItemsMeasurables = allMeasurables[1]
+      if (tabItemsMeasurables.size <= 1) {
+        layout(0, 0) {}
+      } else {
+        // calculate the size
+        val fullWidth = constraints.maxWidth
+        val desiredItemWidth =
+          if (tabItemsMeasurables.any { it.minIntrinsicWidth(oneLineHeight) > fullWidth / 3 }) {
+            fullWidth / 2
+            // this is the maximum width we give the item, if the text is too big for the basic 1/3 of the full width.
+            // If it's still not enough, the text itself gets eclipsed later on
+          } else if (tabItemsMeasurables.size == 2) {
+            fullWidth / 2
+            // if there are only two tabs they always take full width
+          } else {
+            fullWidth / 3
+            // basic
+          }
+
+        // calculate the offsets
+        val howManyLines = (ceil(tabItemsMeasurables.size.toDouble() * desiredItemWidth / fullWidth)).toInt()
+        val howManyItemsInEachLine = fullWidth / desiredItemWidth
+        val fullLayoutCapacity = howManyLines * howManyItemsInEachLine
+        val isLastRowFull = tabItemsMeasurables.size == fullLayoutCapacity
+        val howManyItemsToCenter = if (isLastRowFull || tabItemsMeasurables.size == 2) {
+          0
+        } else {
+          howManyItemsInEachLine - (fullLayoutCapacity - tabItemsMeasurables.size)
+        }
+        val (indicesToCenter, _) = tabItemsMeasurables.indices.partition {
+          it >
+            tabItemsMeasurables.size - 1 - howManyItemsToCenter
+        }
+
+        val mapOfOffsets = mutableMapOf<Int, IntOffset>()
+
+        tabItemsMeasurables.forEachIndexed { index, _ ->
+          val currentRow = if (index <= howManyItemsInEachLine - 1) {
+            0
+          } else if (index <= (2 * howManyItemsInEachLine - 1)) {
+            1
+          } else {
+            2
+          }
+
+          val verticalOffset = oneLineHeight * currentRow
+
+          val horizontalOffset = if (indicesToCenter.contains(index)) {
+            // last line not full, centered
+            calculateHorizontalOffsetForCentered(
+              desiredItemWidth = desiredItemWidth,
+              fullWidth = fullWidth,
+              howManyItemsToCenter = howManyItemsToCenter,
+              currentIndex = index,
+              lastIndex = tabItemsMeasurables.lastIndex,
+            )
+          } else { // full line
+            val adjustedIndex = index - currentRow * howManyItemsInEachLine
```

Comment:
```suggestion
            val adjustedIndex = index - (currentRow * howManyItemsInEachLine)
```
Call me old-fashioned, but I like my parentheses around these just to again make it painfully obvious about what's going on